src/docs/usr-manual/solvers-hypre-ilu.rst

```diff
@@ -10,8 +10,8 @@ hypre-ILU
 
 hypre-ILU is a suite of parallel incomplete LU factorization algorithms featuring dual
 threshold (ILUT) and level-based (ILUK) variants. Its implementation is based on a domain
-decomposition framework for achieving distributed parallelism. It can be used as a
-standalone iterative solver, preconditioner for Krylov subspace methods, or smoother for
+decomposition framework for achieving distributed parallelism. It can be used as
+preconditioners for Krylov subspace methods, or smoothers for
```

Comment:
@liruipeng Shouldn't we mention that it can be used as a standalone solver (even though that's uncommon)? 

Also: the phrase starts with "It", so I think it makes sense to have preconditioner and smoother in the singular form to keep the right verb conjugation

Reply:
I made some grammar changes. Hope it looks better now.

I haven't seen anyone used ILU a "standalone solver". Technically it can; if you set the tolerance to 0, it becomes LU. Maybe it's true that it can used as a crude inner solver of something. But I just felt this could others wrong information about hypre-ILU. Just my 2 cents. 

Thanks!

Reply:
Thanks, Rui Peng! @rfalgout Should we mention, in the documentation, the possibility of using preconditioners in general as standalone solvers (even though this is uncommon)?

Reply:
Absolutely.  It makes more sense to use it as a preconditioner in practice, IMO, but we should let people know.  Thanks!

Reply:
So, I didn't see Rui Peng's comment before I responded.  Is the ILU solver in hypre not already written to be a standalone solver?  I thought that's how we did all of our preconditioning (that was our Babel design anyway).  To ask another way, is it not implemented as x_{k+1} = x_k + M^{-1} A, where M is the ILU approximation?

Reply:
Hi Rob, hypre's ILU is already written to be a standalone solver.

I believe Rui Peng's point was more in the sense: if we mention that it can be used as a standalone solver, people might want to use it in that way. However, that wouldn't be our recommendation (it makes much more sense to use it as a preconditioner or smoother, the same applies to other methods as well, e.g., BoomerAMG, MGR, FSAI...). So, this could possibly confuse users as to what way ILU should be used.

Now, my opinion is: maybe non-advanced users could be confused, but still, I believe we should mention that it is possible to use it as a standalone solver. Lastly, to eliminate confusion, we can add a "Hint" box as we already have in the GPU support section saying that we don't recommend it being used as standalone solver. Something along these lines...

Sorry for the long comment!

Reply:
Why can't we just explicitly say "it can be used as a standalone solver (this is not recommended)"?  I'm not sure a separate hint is needed, but that's okay with me too.  I'm also okay with leaving this out as Rui Peng suggested, but I think it's important that users understand it can technically be used as a standalone solver.

Since preconditioners and smoothers in hypre are all (standalone) solvers (ILU *is* a solver by design), it's a bit redundant to repeat it everywhere, though we need to be sure we say it *somewhere*).  It's probably better for the documentation of individual solvers to discuss recommended use in general.  Maybe this is yet another topic to discuss as a team when we go through our documentation review. :)

Reply:
Saying it as "it can be used as a standalone solver (this is not recommended)" sounds good to me. Thanks!

Reply:
The short statement suggested by @rfalgout and @liruipeng  sounds good to me too. Thanks!

Reply:
Thanks, everyone! Rob raised a good point in his last paragraph. I agree we should talk about this and make a decision, which will be helpful to know when updating the documentation of other methods